PythonPonies/bangazon-storefront
# Ike product types view only

# Description
This pull request allows a user to add a product type, view product types, and see all products for a given product type.

## Number of Fixes
Fresh code

## Related Ticket(s)
List related ticket(s) that apply to the work done in this pull request
user can add product, user can view product types, user can view all products 

## Problem to Solve
Describe the problem(s) this pull request solves.
categorize products and view said products for a given category

## Proposed Changes
Describe the proposed changes.
Add templates and views related to product type
## Expected Behavior
Describe the expected behavior.
Click on products and see the categories of products currently available
Add a product and check to see if the category has been added to the list.
Click on hats and find Nate T Baker's missing hat

## Steps to Test Solution

Delete database and test that all tests pass

1. `git fetch --prune`
2. checkout ike-product-types-view-only
3. run server
4. navigate to products view and ensure category is present (hats)
5. view products within the hat category
6. go back to the product type view and add a product type with the form below the listed items

Thanks!

## Testing

[x ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ x] I certify that all existing tests pass

## Documentation
[x ] There is new documentation in this pull request that must be reviewed..
[ x] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
